### PR TITLE
Make examples extensible

### PR DESCRIPTION
**Pull request by amayvs**
- Restructure existing examples
- Abstract common code to utils
- Improve README files

### CODE REVIEW

**Comment by kpparadigm** [comment]
lgtm

File: examples/Dockerfile [hunk 1]
```diff
@@ -10,4 +10,5 @@ RUN pip install -r /requirements.txt
 COPY *.py /
 # Don't run as root: 65534 is 'nobody' in this image.
 USER 65534:65534
-CMD ["sh", "-c", "python example.py"]
+# Replace `onboarding` with desiered example script.
```
Comment (eugenek1492) [inline examples/Dockerfile:13]: desired

File: examples/README.md [hunk 2]
```diff
@@ -20,11 +12,16 @@ Remember to replace the value of `ETHEREUM_PRIVATE_KEY` below with your own priv
...
 ### Running directly
 
+All commands below are to be run from `examples` directory.
+
 To run it directly, you will need [Python 3.9+](https://www.python.org/downloads/).
 
 ```bash
-pip install -r requirements.txt
-ETHEREUM_PRIVATE_KEY=private_key python example.py
+brew install python3.9 # i.e. ensure python3.9 is available
```
Comment (eugenek1492) [inline examples/README.md:20]: this assumes user uses MacOS? May be leave generic "Make sure you have python V3.9 installed on your computer" ?

File: examples/message_hash.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+types = {
+    "StarkNetDomain": [
+        {"name": "name", "type": "felt"},
+        {"name": "chainId", "type": "felt"},
+        {"name": "version", "type": "felt"},
+    ],
+    "Constant": [
+        {"name": "action", "type": "felt"},
+    ],
+}
```
Comment (eugenek1492) [inline examples/message_hash.py:32]: I see in paradex/amm/api_client_utils.py:auth_message() this code:
```
"types": {
            "StarkNetDomain": [
                {"name": "name", "type": "felt"},
                {"name": "chainId", "type": "felt"},
                {"name": "version", "type": "felt"},
            ],
            "Request": [
                {"name": "method", "type": "felt"},
                {"name": "path", "type": "felt"},
                {"name": "body", "type": "felt"},
                {"name": "timestamp", "type": "felt"},
                {"name": "expiration", "type": "felt"},
            ],
        },
        ```
        but no `Constant` field.
        Both will work I guess ?

**Review by eugenek1492 — APPROVED** [review]
Minor comments, but good to go .

File: examples/message_hash.py (hunk 3 above)
Comment (amayvs) [reply examples/message_hash.py:32]: Both are for different message payloads, see `examples/message_hash.md` for context.

**Review by sv — APPROVED** [review]

**Pull request merged**